vector/src/main/java/im/vector/app/features/settings/devices/v2/DevicesViewModel.kt

```diff
@@ -143,10 +145,24 @@ class DevicesViewModel @AssistedInject constructor(
 
     override fun handle(action: DevicesAction) {
         when (action) {
+            is DevicesAction.VerifyCurrentSession -> handleVerifyCurrentSessionAction()
             is DevicesAction.MarkAsManuallyVerified -> handleMarkAsManuallyVerifiedAction()
         }
     }
 
+    private fun handleVerifyCurrentSessionAction() {
+        viewModelScope.launch {
+            val currentSessionCanBeVerified = checkIfCurrentSessionCanBeVerifiedUseCase.execute()
+            if (currentSessionCanBeVerified) {
+                activeSessionHolder.getSafeActiveSession()?.let { session ->
+                    _viewEvents.post(DevicesViewEvent.SelfVerification(session))
```

Comment:
Passing a whole Session object in a ViewEvent is a bit strange, but it looks like this is actually not used. Please remove it.

Reply:
You are absolutely right. It was a wrong copy-paste of the previous version. I removed it: [87c2b94](https://github.com/vector-im/element-android/pull/7139/commits/87c2b94ff91ac4f8765fe908b5a0924c595334ba)